### PR TITLE
Annotate all local queries for tenant monitoring

### PR DESCRIPTION
**Pull request by gokhangulbiz**
DESCRIPTION: PR description that will go into the change log, up to 78 characters

### CODE REVIEW

File: src/backend/distributed/planner/deparse_shard_query.c [hunk 1]
```diff
@@ -403,8 +403,7 @@ SetTaskQueryIfShouldLazyDeparse(Task *task, Query *query)
 		return;
 	}
 
-	SetTaskQueryString(task, AnnotateQuery(DeparseTaskQuery(task, query),
-										   task->partitionColumn, task->colocationId));
+	SetTaskQueryString(task, DeparseTaskQuery(task, query));
```
Comment (marcocitus) [inline src/backend/distributed/planner/deparse_shard_query.c:406]: we sometimes cache the plan for local execution and reuse it for different distribution key values. Would we still be parsing the annotation in that case?

Comment (gokhangulbiz) [reply src/backend/distributed/planner/deparse_shard_query.c:406]: My intention was to handle cached queries properly. However, upon discussing it with Onder, we observed that this PR doesn't do so. During my testing, I set citus.log_remote_commands to on, which prevented query caching. As our monitoring logic relies on query annotation, we need an actual query to track metrics for cached queries. Therefore, I'm currently exploring alternative options. Do you have any suggestions?

Comment (marcocitus) [reply src/backend/distributed/planner/deparse_shard_query.c:406]: I think you can just use the partitionKeyValue in CitusEndScan

**Comment by gokhangulbiz** [comment]
Moved to https://github.com/citusdata/citus/pull/6797

**Pull request closed without merging**